File: Examples/VisualBasic/ProgrammersGuide/Working-Shapes/SetFillData.vb
Imports Microsoft.VisualBasic
Imports System.IO
Imports Aspose.Diagram

Namespace Shapes
    Public Class SetFillData
        Public Shared Sub Run()
            ' ExStart:SetFillData
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Shapes()

            ' Call the diagram constructor to load diagram from a VSD file
            Dim vdxDiagram As New Diagram(dataDir & "SetFillData.vsd")

            ' Find a particular shape and update its XForm
            For Each shape As Aspose.Diagram.Shape In vdxDiagram.Pages(0).Shapes
                If shape.NameU.ToLower() = "rectangle" AndAlso shape.ID = 1 Then
                    shape.Fill.FillBkgnd.Value = vdxDiagram.Pages(1).Shapes(0).Fill.FillBkgnd.Value
                    shape.Fill.FillForegnd.Value = "#ebf8df"
                End If
            Next shape
            ' save diagram
            vdxDiagram.Save(dataDir & "SetFillData_Out.vsdx", SaveFileFormat.VDX)
            ' ExEnd:SetFillData
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Working-Shapes/SetFillData.cs
using System.IO;
using Aspose.Diagram;
namespace Aspose.Diagram.Examples.CSharp.Shapes
{
    public class SetFillData
    {
        public static void Run()
        {
            // ExStart:SetFillData
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Shapes();

            // Call a Diagram class constructor to load the VSD diagram
            Diagram diagram = new Diagram(dataDir + "SetFillData.vsd");

            // Find a particular shape and update its XForm
            foreach (Aspose.Diagram.Shape shape in diagram.Pages[0].Shapes)
            {
                if (shape.NameU.ToLower() == "rectangle" && shape.ID == 1)
                {
                    shape.Fill.FillBkgnd.Value = diagram.Pages[1].Shapes[0].Fill.FillBkgnd.Value;
                    shape.Fill.FillForegnd.Value = "#ebf8df";
                }
            }
            // Save diagram
            diagram.Save(dataDir + "SetFillData_Out.vsdx", SaveFileFormat.VSDX);
            // ExEnd:SetFillData
        }
    }
}